src/rdkafka_metadata.c

```diff
@@ -584,8 +583,12 @@ rd_kafka_parse_Metadata (rd_kafka_broker_t *rkb,
 
         /* Check if cgrp effective subscription is affected by
          * new topic metadata.
-         * Ignore if this was a broker-only refresh (no topics) */
-        if ((requested_topics || all_topics) && rkb->rkb_rk->rk_cgrp)
+         * Ignore if this was a broker-only refresh (no topics), or
```

Comment:
tangential to this PR (and i may have mentioned it before, recall thinking it before), it might be worth factoring anything that isn't parsing out of a method labelled "parse..". though, the method isn't really doing any harm as is.